lib/mqtt/mqtt.cc

```diff
@@ -68,7 +68,7 @@ namespace
 		NetworkContext_t     networkContext;
 
 		// Lock on which the whole public API synchronizes.
-		FlagLockPriorityInherited lock;
+		RecursiveMutex lock;
```

Comment:
I think that we should add a comment here to explain that CoreMQTT officially discourages recursively calling into the API, and that users recursively calling into the MQTT compartment should be doing it at their own risk (t works on the normal path but we have not asserted that all error paths are safe).

Maybe linking to this issue? https://github.com/FreeRTOS/coreMQTT/issues/278